File: composeApp/src/commonMain/kotlin/com/grateful/deadly/feature/showdetail/components/ShowDetailMenuSheet.kt
package com.grateful.deadly.feature.showdetail.components

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.grateful.deadly.core.design.icons.AppIcon
import com.grateful.deadly.core.design.icons.Render

/**
 * ShowDetailMenuSheet - Menu bottom sheet for show detail actions
 *
 * Follows v2's PlaylistMenuSheet pattern with:
 * - Share option
 * - Choose Recording option
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ShowDetailMenuSheet(
    onShareClick: () -> Unit,
    onChooseRecordingClick: () -> Unit,
    onDismiss: () -> Unit,
    modifier: Modifier = Modifier
) {
    ModalBottomSheet(
        onDismissRequest = onDismiss,
        modifier = modifier
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp)
        ) {
            // Share option
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .clickable {
                        onShareClick()
                        onDismiss()
                    }
                    .padding(vertical = 16.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                AppIcon.Share.Render(
                    size = 24.dp
                )
                Spacer(modifier = Modifier.width(16.dp))
                Text(
                    text = "Share",
                    style = MaterialTheme.typography.bodyLarge
                )
            }

            // Choose Recording option
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .clickable {
                        onChooseRecordingClick()
                        onDismiss()
                    }
                    .padding(vertical = 16.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                AppIcon.LibraryMusic.Render(
                    size = 24.dp
                )
                Spacer(modifier = Modifier.width(16.dp))
                Text(
                    text = "Choose Recording",
                    style = MaterialTheme.typography.bodyLarge
                )
            }

            // Bottom spacing
            Spacer(modifier = Modifier.height(16.dp))
        }
    }
}

File: iosApp/iosApp/Views/ShowDetail/ShowDetailMenuSheet.swift
import SwiftUI
import ComposeApp

/**
 * ShowDetailMenuSheet - Simple menu bottom sheet for show detail actions
 *
 * Matches Compose ShowDetailMenuSheet.kt exactly:
 * - Share option (share icon)
 * - Choose Recording option (library_music icon)
 * - Both dismiss sheet and trigger respective actions
 */

struct ShowDetailMenuSheet: View {
    let onShareClick: () -> Void
    let onChooseRecordingClick: () -> Void
    let onDismiss: () -> Void

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            // Share option
            Button(action: {
                onShareClick()
                onDismiss()
            }) {
                HStack(spacing: 16) {
                    Image(systemName: "square.and.arrow.up")
                        .font(.system(size: 24))
                        .foregroundColor(.primary)
                        .frame(width: 24, height: 24)

                    Text("Share")
                        .font(.body)
                        .foregroundColor(.primary)

                    Spacer()
                }
                .padding(.vertical, 16)
                .padding(.horizontal, 16)
                .contentShape(Rectangle())
            }
            .buttonStyle(.plain)

            Divider()
                .padding(.leading, 56) // Align with text, not icon

            // Choose Recording option
            Button(action: {
                onChooseRecordingClick()
                onDismiss()
            }) {
                HStack(spacing: 16) {
                    Image(systemName: "music.note.list")
                        .font(.system(size: 24))
                        .foregroundColor(.primary)
                        .frame(width: 24, height: 24)

                    Text("Choose Recording")
                        .font(.body)
                        .foregroundColor(.primary)

                    Spacer()
                }
                .padding(.vertical, 16)
                .padding(.horizontal, 16)
                .contentShape(Rectangle())
            }
            .buttonStyle(.plain)

            // Bottom spacing
            Spacer()
                .frame(height: 16)
        }
        .presentationDetents([.height(160)])
        .presentationDragIndicator(.visible)
    }
}

// MARK: - Preview

#Preview {
    ShowDetailMenuSheet(
        onShareClick: { print("Share clicked") },
        onChooseRecordingClick: { print("Choose Recording clicked") },
        onDismiss: { print("Dismissed") }
    )
}
